resolvers.js

```diff
@@ -187,8 +222,65 @@ const resolvers = {
       arrayRemoveItem(databaseDocument.notes, noteId);
       await databaseDocument.save();
 
+      const categoryDocument = await Category.findOne({ _id: categoryId });
+
+      arrayRemoveItem(categoryDocument.notes, noteId);
+      await categoryDocument.save();
+
       return noteDocument;
     },
+
+    deleteDatabaseCategory: async (parent, { databaseId, categoryId }, context) => {
+      assertAuthenticated(context);
+      await verifyDatabaseBelongsToUser(context, databaseId);
+
+      const databaseDocument = await Database.findOne({ _id: databaseId });
+      const categoryDocument = await Category.findOne({ _id: categoryId });
+
+      for (let i = 0; i < categoryDocument.notes; i++) {
+        await Note.findOneAndRemove(
+          { _id: categoryDocument.notes[i] },
+          {
+            useFindAndModify: false
+          }
+        );
+
+        arrayRemoveItem(databaseDocument.notes, categoryDocument.notes[i]);
+        await databaseDocument.save();
+      }
+
+      arrayRemoveItem(databaseDocument.categories, categoryDocument._id);
+      await databaseDocument.save();
```

Comment:
having a save at each iteration of the for loop is generally bad as it incurs a network overhead. batch your updates to the db instead